core/CMakeLists.txt

```diff
@@ -116,7 +116,7 @@ traccc_add_library( traccc_core core TYPE SHARED
   "src/ambiguity_resolution/greedy_ambiguity_resolution_algorithm.cpp" )
 target_link_libraries( traccc_core
   PUBLIC Eigen3::Eigen vecmem::core detray::core detray::detectors
-         traccc::algebra )
+         traccc::algebra ActsCore )
```

Comment:
Can't we please not include the ActsCore in the `traccc::core`?
It will require Boost in compiling traccc core which should be pre-installed.

Reply:
Is it a big issue for `traccc::core` to depend on Boost? If so I am sure we can work around it but Boost is installed basically everywhere already. :thinking:

Reply:
I believe not being depdendent on Boost greatly reduces stress. Upgrading the Boost version has been always painful to me :/ ..

Reply:
Okay, fair enough! I'll work out a way to do this without depending directly on `Acts::Core`. :slightly_smiling_face:

Reply:
Thanks!

Reply:
Boost is one of the most universally used C++ libs, is readily available anywhere (interesting that you can't get it on Perlmutter), and building + upgrading it is ~ trivial, since they essentially never break API, and a CMake-based build possible now.

All in all: having this as a dependency should not be a concern, really. Specifically with the Athena integration, having a single logging system and not multiple ones is beneficial.

We will eventually have to make Acts Core a dependency of traccc anyway when we continue unifying the code bases, so why not do it now? I don't think we should be duplicating work "just because".

Reply:
@beomki-yeo I think the sort-of-official stance on such things at NERSC is to just use a container. I can help you get setup with that if you want, just lmk

Reply:
Sweet. Thanks but I have already asked you to demonstrate that when we prepared the tutorial. I don't think I have heard anything on it from you since then

Reply:
Well you made it work before I got to it ^^ Sorry about that, let's talk next week

Reply:
> We will eventually have to make Acts Core a dependency of traccc anyway when we continue unifying the code bases,

It is opposite If I am correct. What is being developed in ACTS is making traccc a dependency of ACTS.

Reply:
> I  think the sort-of-official stance on such things at NERSC is to just use a container. 

Also please make sure that if you are talking about NERSC standard or ATLAS standard. As far as I can see [here](https://docs.nersc.gov/development/compilers/wrappers/), NERSC is serious in supporting various compilers to general users without relying on containers

Reply:
> It is opposite If I am correct. What is being developed in ACTS is making traccc a dependency of ACTS.

With this change, it depends on the core library, which is what we want so we can eventually share definitions and conventions from ACTS core headers. The traccc plugin effectively makes it a component of the main project.

Reply:
@stephenswat  thanksfully handled the issue so I will resolve the conversation. @paulgessinger We can start discussing whether to include ActsCore as a part of traccc only if you have a concrete idea on it. I am pretty sure that most of Acts EDM or headers will not work in traccc out of the box because they are not GPU friendly. We cannot make `traccc::core` to be dependent on ActsCore just to use a couple of utility headers from it.

Reply:
> a CMake-based build possible now.

Is it possible from https://github.com/boostorg/cmake ? It might be worth trying it out